src/vs/platform/list/browser/listService.ts

```diff
@@ -151,12 +151,13 @@ class WorkbenchOpenController implements IOpenController {
 
 	shouldOpen(event: UIEvent): boolean {
 		if (event instanceof MouseEvent) {
+			const isLeftButton = event.button === 0;
 			const isDoubleClick = event.detail === 2;
-			if (!useSingleClickToOpen(this.configurationService) && !isDoubleClick) {
+			if (isLeftButton && !useSingleClickToOpen(this.configurationService) && !isDoubleClick) {
```

Comment:
Distinguishing between single/double click should only be relevant when using the left button.